app/code/Magento/Deploy/Model/Mode.php

```diff
@@ -83,7 +83,7 @@ class Mode
      * @param OutputInterface $output
      * @param Writer $writer
      * @param Reader $reader
-     * @param MaintenanceMode $maintenanceMode
+     * @param MaintenanceModeEnabler $maintenanceMode
```

Comment:
Such change is backward incompatible and not allowed for a patch releases. You should deprecate old constructor parameter and [add new one](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter).